### PR TITLE
Add changelog for release 4.0

### PR DESCRIPTION
**Pull request by xing-yang**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-4.0.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+# Changelog since v3.0.0
+
+## Breaking Changes
+
+- Update to client/v4 ([#451](https://github.com/kubernetes-csi/external-snapshotter/pull/451), [@xing-yang](https://github.com/xing-yang))
+
+- Move Snapshot APIs to V1. ([#419](https://github.com/kubernetes-csi/external-snapshotter/pull/419), [@xing-yang](https://github.com/xing-yang))
+
+- Update snapshot-controller, CSI snapshotter sidecar, and validation webhook based on v1 apis. ([#425](https://github.com/kubernetes-csi/external-snapshotter/pull/425), [@xing-yang](https://github.com/xing-yang))
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-4.0.md:13]: Note that this requires the v1 CRD to be installed in the cluster.

File: CHANGELOG/CHANGELOG-4.0.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+### Feature
+
+- Add `process_start_time_seconds` metric. ([#387](https://github.com/kubernetes-csi/external-snapshotter/pull/387), [@Jiawei0227](https://github.com/Jiawei0227))
+- The CSI snapshotter has a new `--extra-create-metadata` flag.
+  When set to true, the name and namespace of the source VolumeSnapshot, and the name of the source  VolumeSnapshotContent will be passed as parameters to the CSI driver in CreateSnapshotRequest. ([#375](https://github.com/kubernetes-csi/external-snapshotter/pull/375), [@chrishenzie](https://github.com/chrishenzie))
+- The `--http-endpoint` and `--metrics-path` flags have been added to support metrics for the snapshot-controller. To enable metrics, the user must configure a `--metrics-path` such as `:9102`. The default value for `--metrics-path` is the empty string, meaning this feature is disabled by default. The default value for `--metrics-path` is `/metrics`. ([#409](https://github.com/kubernetes-csi/external-snapshotter/pull/409), [@ggriffiths](https://github.com/ggriffiths))
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-4.0.md:20]: We may need to merge this with the note for #439 below that changed the flag names and added the deprecation notice.

File: CHANGELOG/CHANGELOG-4.0.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+[Documentation](https://kubernetes-csi.github.io)
+
+# Changelog since v3.0.0
+
+## Breaking Changes
+
+- Update to client/v4 ([#451](https://github.com/kubernetes-csi/external-snapshotter/pull/451), [@xing-yang](https://github.com/xing-yang))
+
+- Move Snapshot APIs to V1. ([#419](https://github.com/kubernetes-csi/external-snapshotter/pull/419), [@xing-yang](https://github.com/xing-yang))
```
Comment (msau42) [inline CHANGELOG/CHANGELOG-4.0.md:11]: Adding the v1 API is not breaking because we're still keeping v1beta right?

Comment (xing-yang) [reply CHANGELOG/CHANGELOG-4.0.md:11]: Right. It's only the controller changes which requires this API change.  Let me try to clarify that.

File: CHANGELOG/CHANGELOG-4.0.md (hunk 1 above)
Comment (xing-yang) [reply CHANGELOG/CHANGELOG-4.0.md:13]: Added a note there.

File: CHANGELOG/CHANGELOG-4.0.md (hunk 2 above)
Comment (xing-yang) [reply CHANGELOG/CHANGELOG-4.0.md:20]: 409 is for snapshot-controller while 439 is for the sidecar.  Let me separate them into different categories.

Comment (msau42) [reply CHANGELOG/CHANGELOG-4.0.md:38]: Hmmm I think we want snapshot-controller and the sidecar to also use consistent flag names. Can we quickly put together to make them use similar flag names?

Comment (xing-yang) [reply CHANGELOG/CHANGELOG-4.0.md:38]: They do use similar flag names.

Snapshot-controller just introduced metrics support.  So for snapshot-controller, there's no deprecation as it never used metrics-address.  It just directly uses http-endpoint and metrics-path.

For the sidecar,  metrics-address is deprecated and replaced by http-endpoint.  metrics-path is already there and we are not changing it.

Comment (xing-yang) [reply CHANGELOG/CHANGELOG-4.0.md]: I modified the changelog.  Hope it is more clear.  PTAL.

Comment (msau42) [reply CHANGELOG/CHANGELOG-4.0.md]: ah sorry i'm confusing myself with all the flags :-) Thanks for clarifying

**Pull request merged**